python/lsst/analysis/tools/tasks/associatedSourcesTractAnalysis.py

```diff
@@ -123,6 +161,76 @@ def prepareAssociatedSources(cls, skymap, tract, sourceCatalogs, associatedSourc
 
         return dataFiltered
 
+    @classmethod
+    def applyProperMotionCorrections(cls, dataJoined, properMotionCatalogs, visitTable):
+        """Use proper motion/parallax catalogs to shift positions to median
+        epoch of the visits.
+
+        Parameters
+        ----------
+        dataJoined : `astropy.table.Table`
+            Table containing source positions.
+        properMotionCatalogs: `dict` [`pd.DataFrame`]
+            Dictionary keyed by band with proper motion and parallax catalogs.
+        visitTable : `pd.DataFrame`
+            Table containing the MJDS of the visits.
+
+        Returns
+        -------
+        dataJoined : `astropy.table.Table`
+            Table containing the source positions shifted to the median visit
+            epoch.
+        """
+
+        for band in np.unique(dataJoined["band"]):
+            bandInd = dataJoined["band"] == band
+            bandSources = dataJoined[bandInd].to_pandas()
+            meanRAs = bandSources.groupby("obj_index")["coord_ra"].aggregate("mean")
+            meanDecs = bandSources.groupby("obj_index")["coord_dec"].aggregate("mean")
+
+            bandPMs = properMotionCatalogs[band]
+            with Matcher(meanRAs, meanDecs) as m:
+                idx, i1, i2, d = m.query_radius(
+                    bandPMs["starX"], bandPMs["starY"], 0.2 / 3600, return_indices=True
+                )
+            import ipdb
+
+            ipdb.set_trace()
+            catRAs = np.zeros(len(meanRAs))
+            catDecs = np.zeros(len(meanRAs))
+            pmRAs = np.zeros(len(meanRAs))
+            pmDecs = np.zeros(len(meanRAs))
+            parallaxes = np.zeros(len(meanRAs))
+            catRAs[i1] = bandPMs["starX"][i2]
+            catDecs[i1] = bandPMs["starY"][i2]
+            pmRAs[i1] = bandPMs["starPMx"][i2]
+            pmDecs[i1] = bandPMs["starPMy"][i2]
```

Comment:
The catalog in question was not initially intended to be used by downstream tasks. Converting it to something more user-friendly is on my to-do list, and I'm going to make an actual ticket so that I can cite it here.